sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -99,6 +99,17 @@ object TypeCoercion {
     case (_: TimestampType, _: DateType) | (_: DateType, _: TimestampType) =>
       Some(TimestampType)
 
+    case (t1 @ ArrayType(pointType1, nullable1), t2 @ ArrayType(pointType2, nullable2))
+        if t1.sameType(t2) =>
```

Comment:
why is this `if t1.sameType(t2)` needed?

Reply:
This will make sure the later `get` won't be applied to `None`